stuff2sort/System.py

```diff
@@ -0,0 +1,1150 @@
+                        matchstring = r"^\s*" + re.escape(molecule_name)
+                        match = re.search(matchstring, line, flags=re.MULTILINE)
+                        if match:
+                            found = True
+
+                            for line in ifile:
+                                match = re.search(
+                                    r"^\[\s*atoms\s*\]", line, flags=re.MULTILINE
+                                )
+                                if match:
+                                    break
+                            break
+                        else:
+                            found = False
+                            break
+                    if found:
+                        break
+            for line in ifile:
+                match = re.search(r"^\[", line, flags=re.MULTILINE)
+                if match:
+                    break
+                match = re.search(r"^;", line, flags=re.MULTILINE)
+                if match:
+                    continue
+                match = re.search(
+                    r"^\s*\d+\s+\S+\s+\d+\s+\S+\s+\S+\s+\d+\s+([-]*\d+[\.]*[\d+]*)*",
+                    line,
+                    flags=re.MULTILINE,
+                )
+                if match:
+                    charge_vector.append(float(match.group(1)))
+
+        charge_vector *= molecule_count
+
+        return charge_vector
+
+    def checkformol(self, molecule_name, topology_file):
+
+        '''
+        ------------------------------ \\
+        EFFECT: \\
+        --------------- \\
+        Function Checks If The Molecule (molecule_name) Is Listed In The Topology File \\
+        ------------------------------ \\
+        INPUT: \\
+        --------------- \\
+        molecule_name: str -> Name Of The Molecule \\
+        topology_file: str -> Name Of The Topology File \\
+        ------------------------------ \\
+        RETURN: \\
+        --------------- \\
+        correct: bool \\
+        ------------------------------ \\
+        '''
+
+        with open(topology_file) as ifile:
+            correct = False
+            for line in ifile:
+                match = re.search(r"^;", line, flags=re.MULTILINE)
+                if match:
+                    continue
+                match = re.search(r"^\[\s*moleculetype\s*\]", line, flags=re.MULTILINE)
+                if match:
+                    for line in ifile:
+                        if line.strip() == '' or line.startswith(';'):
+                            continue
+                        else:
+                            matchstring = r"^\s*" + re.escape(molecule_name)
+                            match = re.search(matchstring, line, flags=re.MULTILINE)
+                            if match:
+                                correct = True
+                            break
+                if correct:
+                    break
+
+        return correct
+
+    def read_connectity_from_topology(self, topology_file):
```

Comment:
def read_connectivity_topology would be sufficent here;

file_topology

Return type missing

Reply:
Done